src/libraries/System.Private.CoreLib/src/System/Globalization/Ordinal.cs

```diff
@@ -78,39 +78,73 @@ internal static int CompareStringIgnoreCaseNonAscii(ref char strA, int lengthA,
             return OrdinalCasing.CompareStringIgnoreCase(ref strA, lengthA, ref strB, lengthB);
         }
 
-        private static bool EqualsIgnoreCase_Vector128(ref char charA, ref char charB, int length)
+        private static bool EqualsIgnoreCase_Vector<TVector>(ref char charA, ref char charB, int length)
+            where TVector : struct, ISimdVector<TVector, ushort>
         {
-            Debug.Assert(length >= Vector128<ushort>.Count);
-            Debug.Assert(Vector128.IsHardwareAccelerated);
+            Debug.Assert(length >= TVector.Count);
 
             nuint lengthU = (nuint)length;
-            nuint lengthToExamine = lengthU - (nuint)Vector128<ushort>.Count;
+            nuint lengthToExamine = lengthU - (nuint)TVector.Count;
             nuint i = 0;
-            Vector128<ushort> vec1;
-            Vector128<ushort> vec2;
+            TVector vec1;
+            TVector vec2;
+            TVector loweringMask = TVector.Create(0x20);
+            TVector vecA = TVector.Create('a');
+            TVector vecZMinusA = TVector.Create('z' - 'a');
             do
             {
-                vec1 = Vector128.LoadUnsafe(ref charA, i);
-                vec2 = Vector128.LoadUnsafe(ref charB, i);
+                vec1 = TVector.LoadUnsafe(ref Unsafe.As<char, ushort>(ref charA), i);
+                vec2 = TVector.LoadUnsafe(ref Unsafe.As<char, ushort>(ref charB), i);
```

Comment:
Should we add an `internal static TVector<ushort> LoadUnsafe(ref char source, nuint elementOffset)` helper like we have for `Vector128/256/512` to avoid reintroducing all the `Unsafe.As`s?

Reply:
> Should we add an `internal static TVector<ushort> LoadUnsafe(ref char source, nuint elementOffset)` helper like we have for `Vector128/256/512` to avoid reintroducing all the `Unsafe.As`s?

You will have to make it part of the public ISimdVector interface to work (like, a separate API accepting char), that's why I didn't even try 🙂

Reply:
Ah right, nvm then